dali/pipeline/operators/reader/parser/coco_parser.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_READER_PARSER_COCO_PARSER_H_
+#define DALI_PIPELINE_OPERATORS_READER_PARSER_COCO_PARSER_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "dali/pipeline/operators/reader/parser/parser.h"
+
+namespace dali {
+
+struct Annotation {
+  /*
+   * Currently, only bboxes and labels are supported,
+   * while key points and segmentation masks need to be handled
+   * somehow in the future
+   */
+  Annotation(float x, float y, float w, float h, int category_id)
+    : bbox({x, y, w, h}),
+      category_id(category_id) {}
+
+  // bbox contains 5 values: x, y, W, H, category_id
+  std::vector<float> bbox;
+  int category_id;
+  friend std::ostream& operator<<(std::ostream& os, Annotation& an);
+};
+
+std::ostream& operator<<(std::ostream& os, Annotation& an) {
+  std::vector<float>& bbox = an.bbox;
+  os << "Annotation(category_id=" << an.category_id
+  << ",bbox = [" << bbox[0] << "," << bbox[1]
+  << "," << bbox[2] << "," << bbox[3] << "])";
+  return os;
+}
+
+using AnnotationMap = std::multimap<int, Annotation>;
```

Comment:
It might be nice to document here, what the `int` key in multimap is. I suppose it's image index from COCO set, but there's nothing like knowing for sure